Sources/Sentry/SentrySessionReplayIntegration.m

```diff
@@ -189,30 +205,54 @@ - (void)resumePreviousSessionReplay:(SentryEvent *)event
     }
 
     SentryOnDemandReplay *resumeReplayMaker =
-        [[SentryOnDemandReplay alloc] initWithContentFrom:lastReplayURL.path];
+        [[SentryOnDemandReplay alloc] initWithContentFrom:lastReplayURL.path
+                                          processingQueue:_replayProcessingQueue
+                                         assetWorkerQueue:_replayAssetWorkerQueue
+                                             dateProvider:_dateProvider];
     resumeReplayMaker.bitRate = _replayOptions.replayBitRate;
     resumeReplayMaker.videoScale = _replayOptions.sizeScale;
 
     NSDate *beginning = hasCrashInfo
         ? [NSDate dateWithTimeIntervalSinceReferenceDate:crashInfo.lastSegmentEnd]
         : [resumeReplayMaker oldestFrameDate];
-
     if (beginning == nil) {
         return; // no frames to send
     }
-
-    SentryReplayType _type = type;
-    int _segmentId = segmentId;
-
-    NSError *error;
-    NSArray<SentryVideoInfo *> *videos =
-        [resumeReplayMaker createVideoWithBeginning:beginning
-                                                end:[beginning dateByAddingTimeInterval:duration]
-                                              error:&error];
+    NSDate *end = [beginning dateByAddingTimeInterval:duration];
+
+    // This method is called from a background thread, so we can synchronize the creation of the
+    // video with a dispatch group.
+    __block NSArray<SentryVideoInfo *> *videos;
+    __block NSError *_Nullable error;
+
+    dispatch_group_t group = dispatch_group_create();
```

Comment:
I would appreciate feedback if you asee any issues with this approach for synchronizing the async call